isbg/imaputils.py

```diff
@@ -0,0 +1,373 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+#  imaputils.py
+#  This file is part of isbg.
+#
+#  Copyright 2018 Carles Muñoz Gorriz <carlesmu@internautas.org>
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+#  MA 02110-1301, USA.
+
+"""Imap utils module for isbg - IMAP Spam Begone."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import email          # To easily encapsulated emails messages
+import email.message  # required for typing.TypeVar to work in py3
+import imaplib
+import re             # For regular expressions
+import socket         # to catch the socket.error exception
+import time
+
+from hashlib import md5
+
+from isbg import utils
+from .utils import __
+
+from typing import List, TypeVar, Union
```

Comment:
`typing` is python 3.5+ native library. There is a 2.7 port though. I guess the easiest fix is to add it to the `requirements.txt` file? That way the migration to python3 will be easier.